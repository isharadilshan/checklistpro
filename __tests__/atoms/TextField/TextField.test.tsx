import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TextField from '../../../src/components/atoms/TextField';

configure({adapter: new Adapter()});

const props = {
  label: 'TEST TEXT FIELD',
  required: true,
  labelStyles: {width: 120},
  innerTextInputStyle: {fontSize: 14, color: 'red'},
};

describe('when rendering', () => {
  const wrapper = shallow(<TextField {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
