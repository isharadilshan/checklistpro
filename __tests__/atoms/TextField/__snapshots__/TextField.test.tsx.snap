// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Fragment>
  <Title
    style={
      {
        "fontSize": 16,
        "lineHeight": 18,
        "marginTop": 16,
        "width": 120,
      }
    }
  >
    TEST TEXT FIELD *
  </Title>
  <withTheme(undefined)
    autoCompleteType="off"
    autoCorrect={false}
    mode="outlined"
    placeholderTextColor="black"
    render={[Function]}
    style={
      [
        {
          "height": 50,
        },
      ]
    }
  />
</Fragment>
`;
